pkg/informer/informer.go

```diff
@@ -104,15 +150,16 @@ func NewDynamicDiscoverySharedInformerFactory(
 	handler GVREventHandler,
 	pollInterval time.Duration,
 ) DynamicDiscoverySharedInformerFactory {
-	dsif := dynamicinformer.NewDynamicSharedInformerFactory(dynClient, resyncPeriod)
 	return DynamicDiscoverySharedInformerFactory{
 		workspaceLister: workspaceLister,
 		disco:           disco,
-		dsif:            dsif,
+		dynamicClient:   dynClient,
 		handler:         handler,
 		filterFunc:      filterFunc,
-		gvrs:            sets.NewString(),
+		gvrs:            make(map[schema.GroupVersionResource]struct{}),
```

Comment:
super-nit: a couple lines makes this into a first-class Set (which we can do generically now, I guess) and will improve readability

Reply:
We're still on 1.17

Reply:
But I can wrap this in a set if desired

Reply:
would leave it as it is IMO. Sets are Java'ism.

Reply:
Eh. Less Java-ism and more readability. `foo[thing] = interface{}{}` is strictly worse than `set.Insert(thing)` :)

Feel free to ignore.